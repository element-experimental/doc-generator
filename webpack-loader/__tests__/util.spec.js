const { stripScript, stripStyle, stripTemplate, genInlineComponentText } = require('../util')

// 测试单文件组件
const sfc = `
<template><div>xxxx</div></template>
<script>
export default {
  name: 'sfc'
}
</script>
<style>
div {
  font-size: 12px;
}
</style>
`

describe('test all loader utils', () => {
  test('stripScript', () => {
    const script = stripScript(sfc)
    expect(script).toBe('export default {\n  name: \'sfc\'\n}')
    expect(script).toMatchSnapshot()
  })

  test('stripStyle', () => {
    const style = stripStyle(sfc)
    expect(style).toBe('div {\n  font-size: 12px;\n}')
    expect(style).toMatchSnapshot()
  })

  test('stripTemplate', () => {
    const template = stripTemplate(sfc)
    expect(template).toBe('<template><div>xxxx</div></template>')
    expect(template).toMatchSnapshot()
  })

  test('genInlineComponentText', () => {
    const template = stripTemplate(sfc)
    const script = stripScript(sfc)
    const inlineComponentText = genInlineComponentText(template, script)
    expect(inlineComponentText).toMatchSnapshot()
  })
})
