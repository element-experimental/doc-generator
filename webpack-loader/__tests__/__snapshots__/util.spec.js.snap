// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test all loader utils genInlineComponentText 1`] = `
"(function demo() {
    
    var render = function() {
  var _vm = this
  var _h = _vm.$createElement
  var _c = _vm._self._c || _h
  return _c(\\"div\\", [[_c(\\"div\\", [_vm._v(\\"xxxx\\")])]], 2)
}
var staticRenderFns = []
render._withStripped = true

  
    const democomponentExport = {
  name: 'sfc'
}
    return {
      ...democomponentExport,
      render,
      staticRenderFns
    }
  })()"
`;

exports[`test all loader utils stripScript 1`] = `
"export default {
  name: 'sfc'
}"
`;

exports[`test all loader utils stripStyle 1`] = `
"div {
  font-size: 12px;
}"
`;

exports[`test all loader utils stripTemplate 1`] = `
<template>
  <div>xxxx</div>
</template>
`;
